ProcessCacheTest.shouldRemoveExpiredItem fails intermittently

**Summary**

It was discovered that the ProcessCacheTest.shouldRemoveExpiredTimer has become flaky in the past few days. After debugging the issue I've come to the conclusion that the first assertion fails because of the 10ms cache ttl. 
```java
  void shouldRemoveExpiredItem() throws InterruptedException {
    // given
    configuration.getProcessCache().setExpirationIdleMillis(10L);
    processCache = new ProcessCache(configuration, processFlowNodeProvider, brokerTopologyManager);
    processCache.getCacheItem(1L);
    getCache().cleanUp();
    assertThat(getCacheMap()).hasSize(1); // The test fails here
```
The simple fix is to increase the timeout, but that does not guarantee that it might fail again under extraordinary circumstances sometime in the future. Ultimately, it was decided that the test isn't useful anymore - it's testing the internals of Caffeine, not any Camunda features. 

**Failures**

Here's a CI build where the ProcessCacheTest failed: https://github.com/camunda/camunda/actions/runs/14044783042/attempts/1

And here's the corresponding test output: 

<details><summary>java.lang.AssertionError:  Expected size: but was: 0 in:</summary>
<pre>
{}
	at io.camunda.zeebe.gateway.rest.cache.ProcessCacheTest.shouldRemoveExpiredItem(ProcessCacheTest.java:173)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
</pre>
</details>

**Proposed Fix**

The test case is to be removed as it's no longer providing us any value.

--- a/zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/cache/ProcessCacheTest.java
+++ b/zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/cache/ProcessCacheTest.java
@@ -164,23 +164,6 @@ class ProcessCacheTest {
   }
 
   @Test
-  void shouldRemoveExpiredItem() throws InterruptedException {
-    // given
-    configuration.getProcessCache().setExpirationIdleMillis(10L);
-    processCache = new ProcessCache(configuration, processFlowNodeProvider, brokerTopologyManager);
-    processCache.getCacheItem(1L);
-    getCache().cleanUp();
-    assertThat(getCacheMap()).hasSize(1);
-
-    // when - waiting ttl millis
-    Thread.sleep(10);
-    getCache().cleanUp();
-
-    // then - cache should be empty
-    assertThat(getCacheMap()).isEmpty();
-  }
-
-  @Test
   void shouldRefreshReadItemAndRemoveLeastRecentlyUsed() {
     // given
     configuration.getProcessCache().setMaxSize(2);

